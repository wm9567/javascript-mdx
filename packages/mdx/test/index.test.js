const babel = require('@babel/core')
const mdx = require('../index')
const mdxHastToJsx = require('../mdx-hast-to-jsx')
const fs = require('fs')
const path = require('path')

it('Should output parseable javascript (jsx)', () => {
  const code = mdx('Hello World')
  babel.parse(code, {
    plugins: ["@babel/plugin-syntax-jsx"]
  })
})

it('Should compile fine to snapshot', () => {
  const code = mdx('Hello World')
  expect(code).toMatchSnapshot()
})

it('Should compile sample blogpost to snapshot', () => {
  const fixtureBlogPost = fs.readFileSync(path.join(__dirname,'./fixtures/blog-post.md'))
  const code = mdx(fixtureBlogPost)
  expect(code).toMatchSnapshot()
})

it('Should render blockquote correctly', () => {
  mdx
    .createMdxAstCompiler()
    .use(testResult)
    .processSync('> test\n\n> `test`')

  function testResult () {
    this.Compiler = (tree) => {
      const result = mdxHastToJsx.toJSX(tree.children[0])
      expect(result).toMatchSnapshot()
    }
  }
})
