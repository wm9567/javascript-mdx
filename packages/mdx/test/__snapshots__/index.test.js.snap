// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should compile fine to snapshot 1`] = `
"

export default ({components}) => <MDXTag name=\\"wrapper\\"><MDXTag name=\\"p\\" components={components}>Hello World</MDXTag></MDXTag>"
`;

exports[`Should compile sample blogpost to snapshot 1`] = `
"import { Baz } from './Fixture'
import { Buz } from './Fixture'

export default ({components}) => <MDXTag name=\\"wrapper\\">
<MDXTag name=\\"h1\\" components={components}>Hello, world!</MDXTag>
<MDXTag name=\\"p\\" components={components}>I'm an awesome paragraph.</MDXTag>
<Foo bg='red'>
  <Bar>hi</Bar>
    {hello}
</Foo>
<MDXTag name=\\"p\\" components={components}><MDXTag name=\\"a\\" components={components} parentName=\\"p\\" props={{\\"href\\":\\"https://c8r.imgix.net/028ab8c85da415103cb3b1eb/johno.png\\"}}><MDXTag name=\\"img\\" components={components} parentName=\\"a\\" props={{\\"src\\":\\"https://c8r.imgix.net/028ab8c85da415103cb3b1eb/johno.png\\"}}></MDXTag></MDXTag></MDXTag>
<MDXTag name=\\"p\\" components={components}><MDXTag name=\\"a\\" components={components} parentName=\\"p\\" props={{\\"href\\":\\"https://c8r.imgix.net/028ab8c85da415103cb3b1eb/johno\\"}}>https://c8r.imgix.net/028ab8c85da415103cb3b1eb/johno</MDXTag></MDXTag>
<MDXTag name=\\"pre\\" components={components}><MDXTag name=\\"code\\" components={components} parentName=\\"pre\\">{\`test codeblock
\`}</MDXTag></MDXTag>
<MDXTag name=\\"pre\\" components={components}><MDXTag name=\\"code\\" components={components} parentName=\\"pre\\" props={{\\"className\\":[\\"language-js\\"]}}>{\`module.exports = 'test'
\`}</MDXTag></MDXTag>
<MDXTag name=\\"pre\\" components={components}><MDXTag name=\\"code\\" components={components} parentName=\\"pre\\" props={{\\"className\\":[\\"language-sh\\"]}}>{\`npm i -g foo
\`}</MDXTag></MDXTag></MDXTag>"
`;

exports[`Should render blockquote correctly 1`] = `
"<MDXTag name=\\"blockquote\\" components={components}>
<MDXTag name=\\"p\\" components={components} parentName=\\"blockquote\\">test</MDXTag>
<MDXTag name=\\"p\\" components={components} parentName=\\"blockquote\\"><MDXTag name=\\"inlineCode\\" components={components} parentName=\\"p\\">test</MDXTag></MDXTag>
</MDXTag>"
`;
